src/ripple/shamap/impl/SHAMapTreeNode.cpp

```diff
@@ -76,220 +76,229 @@ SHAMapTreeNode::SHAMapTreeNode(
 }
 
 std::shared_ptr<SHAMapAbstractNode>
-SHAMapAbstractNode::make(
-    Slice const& rawNode,
+SHAMapAbstractNode::makeTransaction(
+    Slice data,
+    std::uint32_t seq,
+    SHAMapHash const& hash,
+    bool hashValid)
+{
+    // FIXME: using a Serializer results in a copy; avoid it?
+    Serializer s(data.begin(), data.size());
+
+    auto item = std::make_shared<SHAMapItem const>(
+        sha512Half(HashPrefix::transactionID, data), s);
+
+    if (hashValid)
+        return std::make_shared<SHAMapTreeNode>(
+            std::move(item), tnTRANSACTION_NM, seq, hash);
+
+    return std::make_shared<SHAMapTreeNode>(
+        std::move(item), tnTRANSACTION_NM, seq);
+}
+
+std::shared_ptr<SHAMapAbstractNode>
+SHAMapAbstractNode::makeTransactionWithMeta(
```

Comment:
`makeTransactionWithMeta` and `makeAccountState` look identical except for the exception messages and `TNTType`, could make a single helper function they both call.

Reply:
Will look, thanks!